lib/galaxy/managers/datasets.py

```diff
@@ -376,7 +381,7 @@ def delete(self, item, flush: bool = True, stop_job: bool = False, **kwargs):
             self.stop_creating_job(item, flush=flush)
         return item
 
-    def purge(self, dataset_assoc, flush=True):
+    def purge(self, item: DI, flush=True, **kwargs) -> DI:
```

Comment:
What is the reason to have `kwargs` here and in many of these functions? :thinking:

Reply:
To maintain the signature with the base class, mypy complains about the signature mismatch otherwise

Reply:
It would be nice to check what is used as parameters, remove those `kwargs`, and add explicit ones at some point, but it is something that can be done later. This is a huge improvement already.